private.tf

```
@@ -17,7 +17,7 @@ locals {
 }
 
 resource "aws_subnet" "private" {
-  count             = local.enabled ? local.availability_zones_count : 0
+  count             = ! local.enabled || var.create_private_subnets == false ? 0 : local.availability_zones_count
```

I prefer to avoid negations where possible, partly in the belief that negations make expressions harder for people to process, but also now that Terraform 0.14 formats negations differently than Terraform 0.13. 
```suggestion
  count             = local.enabled && var.create_private_subnets ? local.availability_zones_count : 0
```